### PR TITLE
Uctbx lrl

### PR DESCRIPTION
**Pull request by dwpaley**
Add python bindings for parts of LatticeRepLib

This builds conditionally only if LatticeRepLib is installed in modules

### CODE REVIEW

**Comment by bkpoon** [comment]
You can ignore the Base tests. If you rebase on top of `master`, those tests might not run.

**Comment by dwpaley** [comment]
> You can ignore the Base tests. If you rebase on top of `master`, those tests might not run.

Thanks!

File: cctbx/uctbx/lrl/match_lattices.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  result = lc.SellingReduceCell(lat, cell, mat)
+  return result
+
+
+inputs = get_input()
+input_readers = []
+for inp in inputs:
+    reader = ext.LRL_ReadLatticeData()
+    reader.CellReader(inp)
+    input_readers.append(reader)
```
Comment (bkpoon) [inline cctbx/uctbx/lrl/match_lattices.py:30]: Inconsistent indentation

File: cctbx/uctbx/lrl/test_match_lattices.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  inputs = [
+      "r 57.98 57.98 57.98 92.02 92.02 92.02",
+      "r 57.98 57.98 57.98 92.02 92.02 92.02",
+      "h 80.36 80.36 99.44 90 90 120",
+      "c 80.95 80.57 57.1 90 90.35 90",
+      "h 80.36 80.36 99.44 90 90 120",
+      "r 57.1 57.1 57.1 89.75 89.75 89.75",
+      ]
+  inp_file = tempfile.NamedTemporaryFile(mode='w')
```
Comment (bkpoon) [inline cctbx/uctbx/lrl/test_match_lattices.py:21]: `inp_file` is not closed and may trigger a warning

**Review by bkpoon — COMMENTED** [review]
Maybe @nksauter and @phyy-nx should also review this since would know more about this code. And are we adding it to Azure Pipelines? It can be run daily if it is not a priority.

Also, please rename `test_match_lattices.py` to `tst_match_lattices.py` so that it is consistent with the test names in `cctbx/regression`.

**Comment by dwpaley** [comment]
Thanks @bkpoon. Requesting Aaron's review. I already discussed with @nksauter offline.

I don't think it's necessary to add this test to Azure. This isn't really functional right now, it requires some dev work on the LatticeRepLib side before we can use it for the intended purpose, which is to resolve pseudomerohedral indexing ambiguity. This PR is essentially a checkpoint so the work doesn't get lost.

**Review by phyy-nx — APPROVED** [review]
I believe the failing test is because `uctbx_lrl` is behind master, so it's not relevant.  So LGTM.

**Pull request merged**